internal/provider/helpers/base_model.go

```diff
@@ -0,0 +1,11 @@
+package helpers
+
+import "github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+
+// BaseModel is embedded in all other types and defines fields
+// common to all Prefect data models.
+type BaseModel struct {
+	ID      customtypes.UUIDValue      `tfsdk:"id"`
+	Created customtypes.TimestampValue `tfsdk:"created"`
+	Updated customtypes.TimestampValue `tfsdk:"updated"`
+}
```

Comment:
This change isn't required for this PR, but saw an opportunity to not repeat these fields. If we like the approach I can come back around in a separate PR and make use of it on the rest of the {re,data}sources.